Tools/Scripts/webkitpy/port/driver.py

```diff
@@ -572,6 +573,8 @@ def cmd_line(self, pixel_tests, per_test_args):
             cmd.append('--show-window')
         if self._port.get_option('accessibility_isolated_tree'):
             cmd.append('--accessibility-isolated-tree')
+        if self._port.get_option('additional_header'):
+            cmd.append("--additional-header '%s'" % self._port.get_option('additional_header'))
```

Comment:
Given this is constructing a list to pass to `subprocess.Popen` or similar, don't we actually want:

```suggestion
            cmd.append("--additional-header")
            cmd.append(self._port.get_option('additional_header'))
```

(compare with the lines just below)

Reply:
> Given this is constructing a list to pass to `subprocess.Popen` or similar, don't we actually want:

After further testing, I see that this change is actually causing WebKitTestRunner to crash in the case where the
`--additional-header` argument has multiple values; for example: `--additional-header='foo=bar hoge=moge'` or whatever. So I’ll need to fix that before this can be merge-worthy

Reply:
> So I’ll need to fix that before this can be merge-worthy

Given this is now resolved, is that fixed?

Reply:
> > So I’ll need to fix that before this can be merge-worthy
> 
> Given this is now resolved, is that fixed?

No — it’s broken now. But it worked before without crashing, in the format-string form — which is surprising, if it’s getting passed to `subprocess.Popen` or similar.

But doing it instead the right way (or what would seem to be the right way) — that is, with multiple append()s — causes a crash (in the server, it seems) when `self._server_process.write(command)` is called.

And it crashes without producing any crash log — no stack trace, no error messages about where it actually fails.

I notice the thing that the existing code is already doing here:

https://github.com/WebKit/WebKit/blob/cb302cd1f44f09c8b330f4c25efb99347b3b8878/Tools/Scripts/webkitpy/port/driver.py#L666-L674

That is, `command` is a string, and that code is constructing it from the arguments using string concatenation, with the arguments separated from each other by `'` apostrophe characters.

And so I guess somewhere in the code it gets by the `self._server_process.write(command)` call, there is some code that parses out that string by splitting it on the `'` apostrophe characters — and it seems like that must be the code where the crash is happening. Not sure, because I haven’t looked there yet — but that’s where I’ll look next.

Anyway, I am guessing that maybe that code doesn’t expect to find a space character in the arguments that get split out from that string?

But so far from debugging, it looks to me like the string it ends up passing in the multiple-append()s case (committed from your suggestion) is actually identical to what it ends up passing from my original format-string form. At least from eyeballing a dump of the strings produced in each case, I can’t see any difference.

Anyway, I remain a bit baffled by it at this point, but as I mentioned, I will next look at the `_server_process.write(command)` code where the `command` string gets handled, and see if that ends up revealing anything more.

Reply:
OK, I’ve fixed the crash. It turns out it’s not actually necessary to append `--additional-header` to the `cmd` value in that `cmd_line` function — the `--additional-header` option has the expected effect without even doing that.

As far as I could see, the only observable effect that appending `--additional-header` there has on anything it to cause `--additional-header` and its value to be appended to the `Command line:` info message that gets emitted by the `print_config` function in the `Printer` logger.

So I instead just directly added some additional info logging to that `print_config` function — to explicitly echo out the `--additional-header` value that’s set in the `self._options.additional_header` value.

And if that all now looks OK, then I think this is otherwise merge-worthy.